### PR TITLE
[NCL-2640] Added client authentication

### PR DESCRIPTION
**Pull request by stanlyDoge**

### CODE REVIEW

File: cli-wrap/src/main/resources/cli/da_cli_script.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
 #!/usr/bin/env python
 import argparse
 import requests
+requests.packages.urllib3.disable_warnings()
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: what this thing does?

File: cli-wrap/src/main/resources/cli/da_cli_script.py [hunk 2]
```diff
@@ -12,14 +14,45 @@
 PRODUCT_VERSION = "PRODUCT_NAME:VERSION"
 STATUS_VALUES   = ['SUPPORTED', 'UNSUPPORTED', 'SUPERSEDED', 'UNKNOWN']
 
-def requests_post(link, json_request):
-    return requests_wrapper(requests.post, link, json_request)
+def get_token(login, pswd):
+    server = "https://pnc-keycloak-vip.host.stage.eng.bos.redhat.com/auth/realms/pncredhat/protocol/openid-connect/token"
+    client_id = "pncdirect"
+    username="pnc-test-user"
+    password="passpass"
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: why is there the testuser & pass?

File: cli-wrap/src/main/resources/cli/da_cli_script.py [hunk 3]
```diff
@@ -12,14 +14,45 @@
...
+    grant_type="password"
+    params = {'grant_type': grant_type,'client_id': client_id,'username': username,'password': password}
+    #params = {'grant_type': 'password','username': username,'password': password}
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: forgotten comment?

File: cli-wrap/src/main/resources/cli/da_cli_script.py [hunk 4]
```diff
@@ -33,20 +66,26 @@ def requests_wrapper(requests_method, link, json_request=None):
 
         # this call throws an exception if the status is 4xx or 5xx
         # we have an exception for 404, since 404 can indicate other
-        # normal stuff to the caller of the method
+        # normal stuff sessionto the caller of the method
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: huh?

File: cli-wrap/src/main/resources/cli/da_cli_script.py [hunk 5]
```diff
@@ -33,20 +66,26 @@ def requests_wrapper(requests_method, link, json_request=None):
...
         if reply.status_code != 404:
             reply.raise_for_status()
-
+
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: whitespace

File: cli-wrap/src/main/resources/cli/listings.py [hunk 6]
```diff
@@ -90,21 +90,7 @@ def matchStatus(status):
...
-      
-    if response.status_code != 200:
-        print("Error")
-        exit()
-    output = response.json()
-    return output
-    #except :
-        #print("Could not connect to server " + pm.da_server)
-        #exit()   
+    return True
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.py]: whitespace at the end of line

File: cli-wrap/src/main/resources/cli/da_cli_script.py (hunk 2 above)
Comment (stanlyDoge) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: just forgotten

File: cli-wrap/src/main/resources/cli/da_cli_script.py (hunk 3 above)
Comment (stanlyDoge) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: yes

File: cli-wrap/src/main/resources/cli/da_cli_script.py (hunk 1 above)
Comment (stanlyDoge) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: it suppresses same warnings - http://stackoverflow.com/questions/27981545/suppress-insecurerequestwarning-unverified-https-request-is-being-made-in-pytho

**Comment by stanlyDoge** [comment]
retest this please

File: cli-wrap/src/main/resources/cli/da_cli_script.py [hunk 7]
```diff
@@ -1,25 +1,59 @@
...
+    r = requests.post(server, params, verify=False)
+    if r.status_code == 200:
+        reply = json.loads(r.content.decode('utf-8'))
+        return( str(reply.get('access_token')))
+    else:
+        return ""
+
+def auth():
+    session = requests.Session() 
+    login = raw_input("Enter login (if empty, system name is used):")
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: What do you thing about showing the default value? Mabye like this:
>Enter login name [sknot]:

Also is there space/newline printed afte the colon?

File: cli-wrap/src/main/resources/cli/da_cli_script.py [hunk 8]
```diff
@@ -1,25 +1,59 @@
...
 import sys
 import json
 import os
+import getpass
 
 
-da_server = os.getenv('DA_SERVER', "pnc-da-cli.cloud.pnc.devel.engineering.redhat.com/da/rest/v-0.4")
+with open('config.json') as config_file:    
+    config = json.load(config_file)
+da_server = os.getenv('DA_SERVER', config["daServer"])
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: verify that da_server is set and not empty and if not, tell user to use configuration or the env variable.

File: cli-wrap/src/main/resources/cli/testsuite.py [hunk 9]
```diff
@@ -2,9 +2,11 @@
 import sys
 import requests
 import os
+import json
 
-DA_MAIN_SERVER = os.getenv('DA_SERVER', "pnc-da-cli.cloud.pnc.devel.engineering.redhat.com/da/rest/v-0.4")
-#DA_MAIN_SERVER="pnc-da-cli.cloud.pnc.devel.engineering.redhat.com/da/rest/v-0.4"
+with open('config.json') as config_file:    
+    config = json.load(config_file)
+DA_MAIN_SERVER = os.getenv('DA_SERVER', config["daServer"])
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/testsuite.py]: fail when DA_MAIN_SERVER is not set.

**Pull request merged**